Classes/Controller/AbstractController.php

```
@@ -68,5 +68,66 @@ class AbstractController extends \TYPO3\CMS\Extbase\Mvc\Controller\ActionControl
 	protected $accessControlService;
 
 
+	/**
+	 * Upload file
+	 */
+	protected function uploadFile(&$fileName, $fileTmpName ) {
+		$basicFileUtility = $this->objectManager->create('TYPO3\CMS\Core\Utility\File\BasicFileUtility');
+		$absFileName = $basicFileUtility->getUniqueName( $fileName, \TYPO3\CMS\Core\Utility\GeneralUtility::getFileAbsFileName('uploads/tx_placements'));
+		$fileInfo = $basicFileUtility->getTotalFileInfo($absFileName);
+		$fileName = $fileInfo['file'];
+		\TYPO3\CMS\Core\Utility\GeneralUtility::upload_copy_move($fileTmpName, $absFileName); 
+	}
+
+	/**
+ 	 * update file property
+ 	 *
+ 	 * @param \TYPO3\CMS\Extbase\DomainObject\AbstractDomainObject $object The object whose proptery should be updated
+ 	 * @param \string $propertyName The property which should be updated
+ 	 * @return void
+ 	 */
+	protected function updateFileProperty($object, $propertyName) {
+		if (\TYPO3\CMS\Extbase\Reflection\ObjectAccess::isPropertyGettable($object, $propertyName) AND
```

We should probably throw an exception if this conditions are not met. (I.e. property is not accessible)